src/source/PeerConnection/PeerConnection.c

```diff
@@ -854,7 +854,7 @@ STATUS addTransceiver(PRtcPeerConnection pPeerConnection, PRtcMediaStreamTrack p
     }
 
     //TODO: Add ssrc duplicate detection here not only relying on RAND()
-    CHK_STATUS(createKvsRtpTransceiver(RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV, pKvsPeerConnection, ssrc,
+    CHK_STATUS(createKvsRtpTransceiver((*ppRtcRtpTransceiver)->direction, pKvsPeerConnection, ssrc,
```

Comment:
You want `pRtcRtpTransceiverInit->direction` and if `pRtcRtpTransceiverInit` is not set you want `RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV`